tests/test_ble_interface_core.py

```
@@ -411,244 +451,1312 @@ def test_ble_interface_defaults_auto_reconnect_disabled(
-        if not thread_errors.empty():
-            where, exc = thread_errors.get_nowait()
-            pytest.fail(f"{where} thread raised {type(exc).__name__}: {exc}")
-    finally:
-        allow_connect.set()
-        if connect_thread.is_alive():
-            connect_thread.join(timeout=1.0)
-        if disconnect_thread.is_alive():
-            disconnect_thread.join(timeout=1.0)
-        iface.close()
+    @contextlib.contextmanager
+    def _swap_target_gate(_target_address: str):
```

Ruff (0.15.4)

[warning] Missing return type annotation for private function `_swap_target_gate`

(ANN202)